ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java

```diff
@@ -107,107 +108,106 @@ public void run() {
-          clearInternalTable(keyedArcticTable, keyedArcticTable.changeTable(), lastTime, execute, metadata, validFiles);
-        } else {
-          throw new IllegalStateException("no pk table, only support mode=all/base");
-        }
-      } else if ("all".equals(mode)) {
-        clearInternalTable(keyedArcticTable, keyedArcticTable.baseTable(), lastTime, execute, metadata, validFiles);
-        clearInternalTable(keyedArcticTable, keyedArcticTable.changeTable(), lastTime, execute, metadata, validFiles);
-      } else {
-        throw new IllegalStateException("only support mode=all/base/change");
-      }
+      clearInternalTableContentsFiles(keyedArcticTable, keyedArcticTable.baseTable(), lastTime, validFiles);
+      clearInternalTableContentsFiles(keyedArcticTable, keyedArcticTable.changeTable(), lastTime, validFiles);
     } else {
-      clearInternalTable(arcticTable, arcticTable.asUnkeyedTable(), lastTime, execute, metadata, validFiles);
+      clearInternalTableContentsFiles(arcticTable, arcticTable.asUnkeyedTable(), lastTime, validFiles);
     }
   }
 
-  private static void clearInternalTable(ArcticTable table, UnkeyedTable internalTable, long lastTime,
-                                         boolean execute, boolean metadata, Set<String> exclude) {
-    if (metadata) {
-      clearInternalTableMetadata(table, internalTable, lastTime, execute);
+  public static void cleanMetadata(ArcticTable arcticTable, long lastTime) {
+    if (arcticTable.isKeyedTable()) {
+      KeyedTable keyedArcticTable = arcticTable.asKeyedTable();
+      clearInternalTableMetadata(keyedArcticTable, keyedArcticTable.baseTable(), lastTime);
+      clearInternalTableMetadata(keyedArcticTable, keyedArcticTable.changeTable(), lastTime);
     } else {
-      clearInternalTableDataFiles(table, internalTable, lastTime, execute, exclude);
+      clearInternalTableMetadata(arcticTable, arcticTable.asUnkeyedTable(), lastTime);
     }
   }
 
-  private static void clearInternalTableDataFiles(ArcticTable table, UnkeyedTable internalTable, long lastTime,
-                                                  boolean execute, Set<String> exclude) {
+  private static void clearInternalTableContentsFiles(ArcticTable table, UnkeyedTable internalTable, long lastTime,
+                                                      Set<String> exclude) {
     int deleteFilesCnt = 0;
     String dataLocation = internalTable.location() + File.separator + DATA_FOLDER_NAME;
     if (table.io().exists(dataLocation)) {
       for (FileStatus fileStatus : table.io().list(dataLocation)) {
         deleteFilesCnt += deleteInvalidDataFiles(table.io(),
             fileStatus,
             lastTime,
-            exclude,
-            execute);
+            exclude);
       }
     }
-    LOG.info("{} total delete[execute={}] {} files", table.id(), execute, deleteFilesCnt);
+    LOG.info("{} total delete {} files", table.id(), deleteFilesCnt);
```

Comment:
Log seems cannot tell the table store information.

Reply:
I modified some logs in the `OrphanFilesCleanService`.

However, I didn't find a convenient way to distinguish `ChangeTable`/`BaseTable` from `UnkeyedTable` by table's name or identifier.